adyen/facade.py

```diff
@@ -144,20 +145,8 @@ def handle_payment_feedback(self, request):
         Validate, process, optionally record audit trail and provide feedback
         about the current payment response.
         """
-        # We must first find out whether this is a redirection or a notification.
-
-        if request.method == 'GET':
-            params = request.GET
-            response_class = PaymentRedirection
-        elif request.method == 'POST':
-            params = request.POST
-            response_class = PaymentNotification
-        else:
-            raise RuntimeError("Only GET and POST requests are supported.")
-
-        # Then we can instantiate the appropriate class from the gateway.
         gateway = get_gateway(request, self.config)
-        response = response_class(gateway, params)
+        response = gateway.get_response()
```

Comment:
I see the idea, but I'm not really happy with the `gateway` depending on `request`. I think we need something a bit different and more isolated to solve this issue.